packages/protocol/test/identity/escrow.ts

```diff
@@ -96,16 +120,100 @@ contract('Escrow', (accounts: string[]) => {
+      assertLogMatches2(receipt.logs[0], {
+        event: 'DefaultTrustedIssuerAdded',
+        args: {
+          trustedIssuer: trustedIssuer1,
+        },
+      })
+    })
+
+    it('should not allow an empty address to be set as a trustedIssuer', async () => {
+      await assertRevertWithReason(
+        escrow.addDefaultTrustedIssuer(NULL_ADDRESS, { from: owner }),
+        "trustedIssuer can't be null"
+      )
+    })
+
+    it('should not allow a trustedIssuer to be added twice', async () => {
+      await escrow.addDefaultTrustedIssuer(trustedIssuer1, { from: owner })
+      await assertRevertWithReason(
+        escrow.addDefaultTrustedIssuer(trustedIssuer1, { from: owner }),
+        'trustedIssuer already in defaultTrustedIssuers'
+      )
+    })
+
+    it('should a second trustedIssuer to be added', async () => {
+      await escrow.addDefaultTrustedIssuer(trustedIssuer1, { from: owner })
+      await escrow.addDefaultTrustedIssuer(trustedIssuer2, { from: owner })
+      assert.deepEqual(await escrow.getDefaultTrustedIssuers(), [trustedIssuer1, trustedIssuer2])
+    })
+  })
+
+  describe('#removeDefaultTrustedIssuer', () => {
+    beforeEach(async () => {
+      await escrow.addDefaultTrustedIssuer(trustedIssuer1, { from: owner })
+    })
+
+    it('allows owner to remove trustedIssuer', async () => {
+      assert.deepEqual(await escrow.getDefaultTrustedIssuers(), [trustedIssuer1])
+      await escrow.removeDefaultTrustedIssuer(trustedIssuer1, 0, { from: owner })
+      assert.deepEqual(await escrow.getDefaultTrustedIssuers(), [])
+    })
+
+    it('reverts if non-owner attempts to remove trustedIssuer', async () => {
+      await assertRevert(
+        escrow.removeDefaultTrustedIssuer(trustedIssuer1, 0, { from: trustedIssuer1 })
+      )
+    })
+
+    it('should emit the DefaultTrustedIssuerRemoved event', async () => {
+      const receipt = await escrow.removeDefaultTrustedIssuer(trustedIssuer1, 0, { from: owner })
+      assertLogMatches2(receipt.logs[0], {
+        event: 'DefaultTrustedIssuerRemoved',
+        args: {
+          trustedIssuer: trustedIssuer1,
+        },
+      })
+    })
+
+    it('should revert if index is invalid', async () => {
+      await assertRevertWithReason(
+        escrow.removeDefaultTrustedIssuer(trustedIssuer1, 1, { from: owner }),
+        'index is invalid'
+      )
+    })
+
+    it('should revert if trusted issuer does not match index', async () => {
+      await assertRevertWithReason(
+        escrow.removeDefaultTrustedIssuer(trustedIssuer2, 0, { from: owner }),
+        'trustedIssuer does not match address found at defaultTrustedIssuers[index]'
+      )
+    })
+
+    it('allows owner to remove trustedIssuer when two are present', async () => {
+      await escrow.addDefaultTrustedIssuer(trustedIssuer2, { from: owner })
+      assert.deepEqual(await escrow.getDefaultTrustedIssuers(), [trustedIssuer1, trustedIssuer2])
```

Comment:
Does deepEqual check ordering?

Reply:
Asking because we want to make sure we get the indexes right

Reply:
Yes I believe it does! DeepEqual should basically compare that the object's properties/values are identical, but that they're not the same object (i.e. deepEqual([1,2,3], [1,2,3]) passes, deepEqual([1,2,3], [2,3,1]) fails, and (===) equal([1,2,3], [1,2,3]) fails